### PR TITLE
LCFS - Refactor ViewTransfer and AddEditTransfer components #455

### PR DESCRIPTION
**Pull request by prv-proton**
- #455 
- [#230 ](https://github.com/bcgov/lcfs/issues/230#issuecomment-2000272987)
- https://github.com/bcgov/lcfs/issues/251#issuecomment-2005129209
- https://github.com/bcgov/lcfs/issues/261#issuecomment-2005109007
- https://github.com/bcgov/lcfs/issues/326#issuecomment-2005179642

### CODE REVIEW

File: frontend/src/constants/statuses.js [hunk 1]
```diff
@@ -1,12 +1,19 @@
...
-  draft: 'Draft',
-  deleted: 'Deleted',
-  sent: 'Sent',
-  submitted: 'Submitted',
-  recommended: 'Recommended',
-  unregistered: 'Unregistered',
-  registered: 'Registered',
-  suspended: 'Suspended',
-  canceled: 'Canceled'
+export const TRANSFER_STATUS = {
```
Comment (AlexZorkin) [inline frontend/src/constants/statuses.js:1]: Let's update these to be plural so we know they are options lists. TRANSFER_STATUSES, ORGANIZATION_STATUSES

export const ORGANIZATION_STATUSES = {
  UNREGISTERED: 'Unregistered',
  REGISTERED: 'Registered',
  SUSPENDED: 'Suspended',
  CANCELED: 'Canceled'
};

// Function to get all organization statuses as an array
function getAllOrganizationStatuses() {
  return Object.values(ORGANIZATION_STATUSES);
}

// Usage
const statusArr = getAllOrganizationStatuses();

File: frontend/src/utils/cellRenderers.jsx [hunk 2]
```diff
@@ -52,8 +53,8 @@ export const StatusRenderer = (props) => {
 
 export const OrgStatusRenderer = (props) => {
   const location = useLocation()
-  const statusArr = ['Registered', 'Unregistered', 'Suspended', 'Canceled']
-  const statusColorArr = ['success', 'info', 'warning', 'error']
+  const statusArr = Object.values(ORGANIZATION_STATUS)
```
Comment (AlexZorkin) [inline frontend/src/utils/cellRenderers.jsx:56]: use the previously mentioned function
const statusArr = getAllOrganizationStatuses();

File: frontend/src/views/Transfers/components/TransferView.jsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+          {t('transfer:transfers')}
+          <b>{quantity}</b>
+          {t('transfer:complianceUnitsTo')} <b>{toOrganization}</b>
+          {t('transfer:for')}
+          <b>${decimalFormatter({ value: pricePerUnit })}</b>
+          {t('transfer:complianceUnitsPerTvo')}
+          <b>${decimalFormatter(totalValue)}</b> CAD.
+        </Typography>
+      </BCBox>
+      -- demo data --
```
Comment (AlexZorkin) [inline frontend/src/views/Transfers/components/TransferView.jsx:62]: Let's remove these demo data markers. just hide the CommentList, AttachmentList, and transactionHistory. We have prs coming for two of those 3.

**Review by AlexZorkin — CHANGES_REQUESTED** [review]
The validation classes you've created on the backend are great. Those really help with the separation of concerns idea with the repository pattern. Just a few minor changes otherwise.

File: backend/lcfs/web/api/organizations/repo.py [hunk 4]
```diff
@@ -187,3 +187,18 @@ async def get_organization_attorney_address(self, organization_attorney_address_
...
+    @repo_handler
+    async def is_registered_for_transfer(self, organization_id: int):
+        """
+        Check if an organization is registered in the database.
+        """
+        result = await self.db.scalar(
+            select(Organization.organization_id).where(
+                and_(
+                    Organization.organization_id == organization_id,
+                    Organization.organization_status_id == 2,
```
[the remark this answers is not in the document]
Comment (prv-proton) [reply backend/lcfs/web/api/organizations/repo.py:200]: updated.

File: frontend/src/constants/statuses.js (hunk 1 above)
Comment (prv-proton) [reply frontend/src/constants/statuses.js:1]: updated as per comments.

File: frontend/src/views/Transfers/components/TransferView.jsx (hunk 3 above)
Comment (prv-proton) [reply frontend/src/views/Transfers/components/TransferView.jsx:62]: removed and commented.

**Comment by prv-proton** [comment]
Fixed all the reviews. Please re-review.

**Pull request merged**